proxy-node-gateway/src/test/java/uk/gov/ida/notification/apprule/rules/TestSamlResponseExceptionMapperRule.java

```diff
@@ -0,0 +1,108 @@
+package uk.gov.ida.notification.apprule.rules;
+
+import io.dropwizard.testing.junit.DropwizardClientRule;
+import io.dropwizard.testing.junit.ResourceTestRule;
+import org.glassfish.jersey.client.ClientProperties;
+import org.glassfish.jersey.test.grizzly.GrizzlyWebTestContainerFactory;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.slf4j.MDC;
+import uk.gov.ida.notification.SamlFormViewBuilder;
+import uk.gov.ida.notification.exceptions.mappers.ErrorPageExceptionMapper;
+import uk.gov.ida.notification.exceptions.mappers.ExceptionToSamlErrorResponseMapper;
+import uk.gov.ida.notification.exceptions.mappers.GenericExceptionMapper;
+import uk.gov.ida.notification.proxy.TranslatorProxy;
+import uk.gov.ida.notification.session.GatewaySessionData;
+import uk.gov.ida.notification.session.storage.SessionStore;
+import uk.gov.ida.notification.shared.ProxyNodeMDCKey;
+import uk.gov.ida.notification.views.SamlFormView;
+
+import javax.annotation.RegEx;
+import javax.ws.rs.core.Response;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.doCallRealMethod;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class TestSamlResponseExceptionMapperRule {
+
+    private static final SamlFormViewBuilder samlFormViewBuilder = mock(SamlFormViewBuilder.class);
+    private static final TranslatorProxy translatorProxy = mock(TranslatorProxy.class);
+    private static final SessionStore sessionStore = mock(SessionStore.class);
+
+    @Before
+    public void before() {
+        when(sessionStore.getSession(any(String.class))).thenReturn(new GatewaySessionData("HubRequestId","EidasRequestId","EidasDestination","EidasConnectorPublicKey","EidasRelayState"));
+        String nullString = null;
```

Comment:
This is fine but it's okay to just use `null` as an arg in a method call, especially in a test

Reply:
In this case, passing in null is ambiguous as far as the compiler is concerned as the second argument can be either a String or a Response depending on the overload so you have to use a variable of type String set to null to stop the compiler from getting confused.

Reply:
Oh, right, fair enough then and good job getting around that!